### PR TITLE
Feature/reduce orders footprint

### PR DESCRIPTION
**Pull request by jeevanjoseph**
Enables Metrics as well as a new default grafana dashboard. 

- Spring boot apps now expose actuator based metrics. 
- Deployment annotations pickup pods to be scraped and update prometheus scrape config
- default dashboard for visualizing spring boot metrics. 
- Orders build updated to use new containers (old workarounds removed.)
- updated jdbc version.

### CODE REVIEW

File: src/orders/Dockerfile [hunk 1]
```diff
@@ -4,29 +4,16 @@ ARG VERSION="0.0.1-SNAPSHOT"
 
 # ------------
 # Stage 1 : Setup the build environment
-FROM maven:3.6-jdk-8 as buildenv
-ARG GRADLE_VERSION=6.5
-ENV GRADLE_HOME /opt/gradle
+FROM gradle:6.5 as buildenv
```
Comment (mvandervliet) [inline src/orders/Dockerfile:7]: 👍

File: deploy/complete/helm-chart/mushop/charts/orders/templates/orders-deployment.yaml [hunk 2]
```diff
@@ -20,6 +20,9 @@ spec:
         mockmode: {{ .Values.global.mock.service | quote }}
       annotations:
         sidecar.istio.io/rewriteAppHTTPProbers: "true"
+        prometheus.io/scrape: "true"
+        prometheus.io/path: /actuator/prometheus
+        prometheus.io/port: "80"
```
Comment (junior) [inline deploy/complete/helm-chart/mushop/charts/orders/templates/orders-deployment.yaml:25]: The actuator defaults to port 80?

File: deploy/complete/helm-chart/mushop/charts/orders/templates/orders-deployment.yaml [hunk 3]
```diff
@@ -20,6 +20,9 @@ spec:
         mockmode: {{ .Values.global.mock.service | quote }}
       annotations:
         sidecar.istio.io/rewriteAppHTTPProbers: "true"
+        prometheus.io/scrape: "true"
+        prometheus.io/path: /actuator/prometheus
```
Comment (junior) [inline deploy/complete/helm-chart/mushop/charts/orders/templates/orders-deployment.yaml:24]: Could't you also use the actuators to do liveness and readiness probes?

Comment (jeevanjoseph) [reply deploy/complete/helm-chart/mushop/charts/orders/templates/orders-deployment.yaml:24]: Yes - if fact it’s enabled. But the deployment template is not setup to use it, will add the liveness and readiness probes to the template

File: deploy/complete/helm-chart/mushop/charts/orders/templates/orders-deployment.yaml (hunk 2 above)
Comment (jeevanjoseph) [reply deploy/complete/helm-chart/mushop/charts/orders/templates/orders-deployment.yaml:25]: Yes - defaults to whatever port the web server is running on - jetty in this case and it runs on 80.

**Comment by jeevanjoseph** [comment]
@junior can you please 👀 once more ?
Added the probes are realized `startupProbe` cannot  be added before 1.16 (I was using 1.16.5 locally)

**Pull request merged**